### PR TITLE
Deprecate CC disk + cpuUtilization capacity properties

### PR DESCRIPTION
**Pull request by kyguy**
### Type of change

- Enhancement / new feature

### Description
Addresses #6293 

Deprecates the following properties in the Kafka resource:

-  `spec.cruiseControl.capacity.cpu`
-  `spec.cruiseControl.capacity.disk`

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [x] Make sure all tests pass
- [x] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [x] Reference relevant issue(s) and close them after merging
- [x] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

**Comment by kyguy** [comment]
Hey @scholzj for deprecating these properties I tried following the format of some past deprecation PRs, could you take a look to see if I am missing anything?

**Review by scholzj — COMMENTED** [review]
I think this is good. Add it to the CHANGELOG.md => to the section for deprecations and removals.

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,7 @@
 ### Changes, deprecations and removals
 
 * The `type` field in `ListenerStatus` has been deprecated and will be removed in the future.
+* The `disk` and `cpuUtilization` fields in the `spec.cruiseControl.capacity` section of the Kafka resource have been deprecated and will be removed in the future.
```
Comment (scholzj) [inline CHANGELOG.md:15]: Maybe you should also clarify whether they are still used for anything or ignored.

**Review by scholzj — APPROVED** [review]
One nit. LGTM otherwise. Should be reviewed by SMEs.

File: api/src/main/java/io/strimzi/api/kafka/model/balancing/BrokerCapacity.java [hunk 2]
```diff
@@ -42,6 +43,8 @@
     private String outboundNetwork;
     private Map<String, Object> additionalProperties = new HashMap<>(0);
 
+    @Deprecated
+    @DeprecatedProperty(description = "The Cruise Control disk capacity setting has been removed")
```
Comment (ppatierno) [inline api/src/main/java/io/strimzi/api/kafka/model/balancing/BrokerCapacity.java:47]: AFAIU they are deprecated and ignored but not removed so the user can still set them but the operator won't use them. Is that right? Should we clarify in this description?

File: api/src/main/java/io/strimzi/api/kafka/model/balancing/BrokerCapacity.java [hunk 3]
```diff
@@ -55,6 +58,8 @@ public void setDisk(String disk) {
         this.disk = disk;
     }
 
+    @Deprecated
+    @DeprecatedProperty(description = "The Cruise Control CPU capacity setting has been removed")
```
Comment (ppatierno) [inline api/src/main/java/io/strimzi/api/kafka/model/balancing/BrokerCapacity.java:62]: Ditto as above

**Review by ppatierno — APPROVED** [review]

**Comment by kyguy** [comment]
@scholzj I just fixed the failing test, could you run the regression tests again

File: documentation/modules/appendix_crds.adoc [hunk 4]
```diff
@@ -1352,9 +1352,9 @@ Used in: xref:type-CruiseControlSpec-{context}[`CruiseControlSpec`]
 [options="header"]
 |====
 |Property                |Description
-|disk             1.2+<.<a|Broker capacity for disk in bytes. Use a number value with either standard Kubernetes byte units (K, M, G, or T), their bibyte (power of two) equivalents (Ki, Mi, Gi, or Ti), or a byte value with or without E notation. For example, 100000M, 100000Mi, 104857600000, or 1e+11.
+|disk             1.2+<.<a|*The `disk` property has been deprecated.* The Cruise Control disk capacity setting has been deprecated, is ignored, and will be removed in the future Broker capacity for disk in bytes. Use a number value with either standard Kubernetes byte units (K, M, G, or T), their bibyte (power of two) equivalents (Ki, Mi, Gi, or Ti), or a byte value with or without E notation. For example, 100000M, 100000Mi, 104857600000, or 1e+11.
```
Comment (tomncooper) [inline documentation/modules/appendix_crds.adoc:1355]: Should we mention that it is now automatically generated?

**Review by tomncooper — APPROVED** [review]
LGTM

**Comment by kyguy** [comment]
@scholzj This is ready for merge! I will add more docs in a later PR if Tom wants more!

**Pull request merged**